internal/db/fetcher/indexer_iterators.go

```
@@ -594,6 +593,72 @@ func (f *IndexFetcher) determineFieldFilterConditions() ([]fieldFilterCond, erro
 	return result, nil
 }
 
+// determineJSONFilterCondition determines the condition and its corresponding operation for a
+// JSON filter condition.
+// It mutates the given condition to make it match the filter value and JSON path so that
+// it can be used to fetch the indexed data.
+func determineJSONFilterCondition(cond *fieldFilterCond, filterVal any, jsonPath client.JSONPath) error {
+	var jsonVal client.JSON
+	var err error
+	if isArrayCondition(cond.op) {
+		subCondMap := filterVal.(map[connor.FilterKey]any)
+		for subKey, subVal := range subCondMap {
+			cond.arrOp = cond.op
+			cond.op = subKey.(*mapper.Operator).Operation
+			jsonVal, err = client.NewJSONWithPath(subVal, jsonPath)
```

todo: This flow is a little confusing and looks like you are silently discarding the error.  Perhaps the following would be a little clearer?
```
jsonVal, err = client.NewJSONWithPath(subVal, jsonPath)
if err != nil {
  return err
}
cond.val = client.NewNormalJSON(jsonVal)
```

---

dont